Octokit.Tests/Clients/BlobClientTests.cs

```diff
@@ -122,7 +122,7 @@ public void CanDeserializeBlob()
             Assert.NotNull(response.Body);
             Assert.Equal(blobResponseJson, (string)response.HttpResponse.Body);
             Assert.Equal(100, response.Body.Size);
-            Assert.Equal(EncodingType.Utf8, response.Body.Encoding);
+            Assert.Equal(EncodingType.Utf8, response.Body.Encoding.ParsedValue);
```

Comment:
I guess another way to tackle this could be to change the equality operator between 2 `StringEnum<TEnum>` to compare their `ParsedValue` (if both of them succesfully can `TryParse()`) and if not, fall back to comparing the underlying string value as it currently does.

This would mean in the case of

`{ Value = "utf8", ParsedValue = EncodingType.Utf8 }` (implicitly converted from `EncodingType.Utf8`)
`{ Value = "utf-8", ParsedValue = EncodingType.Utf8 }` (deserialized from api response "utf-8")

They would correctly be seen as "equal".

Thoughts?

Although more work to change 42 enums, I think Im keen on my earlier suggestion of being explicit with `[Parameter()]` attributes on every enum member, enforced via convention test, and relied on by the deserializer